openmdao/docs/openmdao_book/features/core_features/working_with_components/distributed_components.ipynb

```diff
@@ -776,7 +776,7 @@
     "                    deriv = np.tile(dg_dx, ny).reshape((ny, nx)).dot(d_inputs['in_dist'])\n",
     "                    if MPI and self.comm.size > 1:\n",
     "                        # In Fwd, allreduce the result of the dot product with the subjac.\n",
-    "                        deriv_sum = np.zeros(deriv.size)\n",
+    "                        deriv_sum = np.zeros_like(deriv)\n",
```

Comment:
maybe put a little comment here to say why it's good to avoid assuming dtype=float

Reply:
I added a somewhat general comment about it.  I think if I go into any detail about complex-stepping across an derivative calculation, it might cause some confusion.